Source/BrewInfo.spoon/init.lua

```diff
@@ -0,0 +1,137 @@
+--- === BrewInfo ===
+---
+--- Display pop-up with Homebrew Formula info, or open their URL
+---
+--- Download: [https://github.com/Hammerspoon/Spoons/raw/master/Spoons/BreInfo.spoon.zip](https://github.com/Hammerspoon/Spoons/raw/master/Spoons/BrewInfo.spoon.zip)
+---
+--- You can bind keys to automatically display the output of `brew
+--- info` of the currently-selected package name, or to open its
+--- homepage. I use it to quickly explore new packages from the output
+--- of `brew update`.
+
+local mod={}
+mod.__index = mod
+
+-- Metadata
+mod.name = "BrewInfo"
+mod.version = "1.0"
+mod.author = "Diego Zamboni <diego@zzamboni.org>"
+mod.homepage = "https://github.com/Hammerspoon/Spoons"
+mod.license = "MIT - https://opensource.org/licenses/MIT"
+
+mod.key_show_brew_info = nil
+mod.key_open_brew_url = nil
+
+--- BrewInfo.brew_info_delay_sec
+--- Variable
+--- An integer specifying how long the alerts generated by BrewInfo will stay onscreen
+mod.brew_info_delay_sec = 3
+
+--- BrewInfo.brew_info_style
+--- Variable
+--- A table in conformance with the [hs.alert.defaultStyle](http://www.hammerspoon.org/docs/hs.alert.html#defaultStyle[]) format that specifies the style used by the alerts. Default value: `{ textFont = "Courier New", textSize = 14, radius = 10 }`
+mod.brew_info_style = {
+   textFont = "Courier New",
+   textSize = 14,
+   radius = 10
+}
+
+-- Internal function to get the currently selected text
+function current_selection()
+   local elem=hs.uielement.focusedElement()
+   if elem then
+      return elem:selectedText()
+   else
+      return nil
+   end
+end
+
+-- Internal method to show an alert in the configured style
+function mod:show(text)
+   hs.alert.show(text, self.brew_info_style, self.brew_info_delay_sec)
+   return self
+end
+
+--- BrewInfo:showBrewInfo(pkg)
+--- Method
+--- Displays an alert with the output of `brew info <pkg>`
+---
+--- Parameters:
+---  * pkg - name of the package to query
+function mod:showBrewInfo(pkg)
+   if pkg and pkg ~= "" then
+      local info, st, t, rc=hs.execute("/usr/local/bin/brew info " .. pkg)
+      if st == nil then
+         info = "No information found about formula '" .. pkg .. "'!"
+      end
+      self:show(info)
+   else
+      self:show("No package selected.")
+   end
+   return self
+end
+
+--- BrewInfo:showBrewInfoCurSel()
+--- Method
+--- Display `brew info` using the selected text as the package name
```

Comment:
By convention, I always add `Parameters:` and `Returns:` even if they just have `* None` in them (although in this case, it does return something.